lib/Sema/TypeCheckConcurrency.cpp

```diff
@@ -774,22 +773,36 @@ namespace {
 
       // It is within a nonisolated context.
       NonIsolatedContext,
+
+    };
+
+    enum PotentiallyIsolatedKind {
+      PotentiallyIsolated,
+      DefinitelyNotIsolated
     };
```

Comment:
Let's be more specific with the naming around this. What does it mean to be "potentially" isolated vs "definitely not" isolated, according to this enum for the `isolatedBase` member? If that's too hard to describe, what is the consequence of it being potentially isolated?

My guess is this tracks whether the reference is for distributed actor and it might be following the special "known local" semantics, where it's non-throwing?

Reply:
Hmm so that's hard for me since I also don't have a good name for it and just used how it was called internally in the impl:


```
bool isPotentiallyIsolated = false;
      if (!var) {
        isPotentiallyIsolated = false;
      } else if (auto param = dyn_cast<ParamDecl>(var)) {
        isPotentiallyIsolated = param->isIsolated();
      } else if (var->isSelfParamCapture()) {
        // Find the "self" parameter that we captured and determine whether
        // it is potentially isolated.
        for (auto dc = var->getDeclContext(); dc; dc = dc->getParent()) {
          if (auto func = dyn_cast<AbstractFunctionDecl>(dc)) {
            if (auto selfDecl = func->getImplicitSelfDecl()) {
              isPotentiallyIsolated = selfDecl->isIsolated();
              break;
            }
          }

          if (dc->isModuleScopeContext() || dc->isTypeContext())
            break;
        }
      }
```

So this _really_ represents... if the base is something that _can_ be isolated to...? 🤔

Reply:
Made be the isPotentiallyIsolated and documented what it means :)